pkg/distributor/push.go

```diff
@@ -100,14 +100,22 @@ func Handler(
 
 type distributorMaxWriteMessageSizeErr struct {
 	actual, limit int
+	id            globalerror.ID
 }
 
 func (e distributorMaxWriteMessageSizeErr) Error() string {
 	msgSizeDesc := fmt.Sprintf(" of %d bytes", e.actual)
+	var flag string
 	if e.actual < 0 {
 		msgSizeDesc = ""
 	}
-	return globalerror.DistributorMaxWriteMessageSize.MessageWithPerInstanceLimitConfig(fmt.Sprintf("the incoming push request has been rejected because its message size%s is larger than the allowed limit of %d bytes", msgSizeDesc, e.limit), "distributor.max-recv-msg-size")
+	if e.id == globalerror.DistributorMaxOtelDecompressedWriteMessageSize {
+		flag = "distributor.max_otel_decompressed_recv_msg_size"
```

Comment:
`-distributor.max_otel_decompressed_recv_msg_size` is incorrect. To avoid mismatch, extract the flag name to a string constant and use it in all places where needed.

Same comment about extracting flag name to constant applies to `distributor.max-recv-msg-size`.

Reply:
renaming and const is done in commit https://github.com/grafana/mimir/commit/1ba1bcb4854d016dfada97077fc1696ea23c84e3